src/controllers/py.controller.ts

```diff
@@ -0,0 +1,163 @@
+import { v4 as uuid4 } from 'uuid';
+import { spawn } from "child_process";
+import {
+    Contributor,
+    MetadataField,
+    NotebookCellExecution,
+    NotebookController,
+    OnControllerInfo,
+} from "../core/controller";
+import * as UniqueFileName from 'uniquefilename';
+import path = require('path');
+import { tmpdir } from 'os';
+import * as fs from 'fs';
+
+
+export class PyController extends NotebookController implements OnControllerInfo {
+    private static readonly _supportedLanguages: string[] = ['python'];
+    private static readonly _detail: string = 'Run python code on local machine';
+    private static readonly _description: string = 'Local python execution';
+    private static readonly _controllerId: string = 'py-local';
+    private static readonly _notebookType: string = 'pythonbook';
+    private static readonly _label: string = 'Python Local';
+    private static readonly _execMeta: string = 'execution';
+    private static readonly _defaultExecutionCmd: string = `python`;
+    private static readonly _execFileArg: string = 'xprog';
+
+    constructor() {
+        super(
+            PyController._controllerId,
+            PyController._notebookType,
+            PyController._label,
+        );
+    }
+    cellMetadata(): MetadataField[] | undefined {
+        return [
+            {
+                key: PyController._execMeta,
+                default: PyController._defaultExecutionCmd,
+                description: `where {${PyController._execFileArg}} is the path to the temporary file to execute`
+            }
+        ];
+    }
+    notebookMetadata(): MetadataField[] | undefined {
+        return;
+    }
+
+    public supportedLanguages(): string[] | undefined {
+        return PyController._supportedLanguages;
+    }
+
+    public supportsExecutionOrder(): boolean | undefined {
+        return false;
+    }
+
+    public detail(): string | undefined {
+        return PyController._detail;
+    }
+
+    public description(): string | undefined {
+        return PyController._description;
+    }
+
+    public contributors(): Contributor[] | undefined {
+        return [
+            {
+                name:  'Artsem Hutarau',
+                email: 'gutorov.artem@yandex.ru',
+                url:   'https://github.com/MonkeyBuisness'
+            },
+            {
+                name: 'Raman H',
+                email: [
+                    'e', 'm', '.', 'm',
+                    'p', '@', 'h', '.',
+                    'n', 'a', 'm', 'a',
+                    'r'
+                ].reverse().join(''),
+                url: 'https://github.com/duckyou',
+            },
+        ];
+    }
+
+    public icon(): string | undefined {
+        return 'py-local.png';
```

Comment:
Don't forget to add `py-local.png`(256x256 px) to the [resources/kernels/icons](https://github.com/MonkeyBuisness/cellementary-extension/tree/master/resources/kernels/icons) folder